llama_index/vector_stores/rocksetdb.py

```diff
@@ -0,0 +1,296 @@
+from __future__ import annotations
+from os import getenv
+from time import sleep
+from enum import Enum
+from typing import List, Any, Optional, TypeVar, Type
+from types import ModuleType
+from llama_index.vector_stores.types import (
+    NodeWithEmbedding,
+    VectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+from llama_index.vector_stores.utils import (
+    node_to_metadata_dict,
+    metadata_dict_to_node,
+    DEFAULT_TEXT_KEY,
+    DEFAULT_EMBEDDING_KEY,
+)
+
+T = TypeVar("T", bound="RocksetVectorStore")
+
+
+def _get_rockset() -> ModuleType:
+    """Gets the rockset module and raises an ImportError if
+    the rockset package hasn't been installed
+
+    Returns
+        rockset module (ModuleType)
+    """
+    try:
+        import rockset
+    except ImportError:
+        raise ImportError("Please install rockset with `pip install rockset`")
+    return rockset
+
+
+def _get_client(
+    api_key: Optional[str], api_server: Optional[str], client: Optional[Any]
+) -> Any:
+    """Returns the passed in client object if valid, else
+    constructs and returns one.
+
+    Returns
+        The rockset client object (rockset.RocksetClient)
+    """
+    rockset = _get_rockset()
+    if client:
+        if not type(client) is rockset.RocksetClient:
+            raise ValueError("Parameter `client` must be of type rockset.RocksetClient")
+    elif not api_key and not getenv("ROCKSET_API_KEY"):
+        raise ValueError(
+            "Parameter `client`, `api_key` or env var `ROCKSET_API_KEY` must be set"
+        )
+    else:
+        client = rockset.RocksetClient(
+            api_key=api_key or getenv("ROCKSET_API_KEY"),
+            host=api_server or getenv("ROCKSET_API_SERVER"),
+        )
+    return client
+
+
+class RocksetVectorStore(VectorStore):
+    stores_text: bool = True
+    is_embedding_query: bool = True
```

Comment:
does Rockset require flat metadata? We should add the attribute here, similar to other vector stores

Reply:
Added in https://github.com/jerryjliu/llama_index/pull/7111/commits/ee9dfcd2d637f15a2612f19876741a41dccde078.